server/src/main/java/org/apache/iotdb/db/mpp/operator/schema/CountMergeOperator.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.db.mpp.operator.schema;
+
+import org.apache.iotdb.db.mpp.operator.Operator;
+import org.apache.iotdb.db.mpp.operator.OperatorContext;
+import org.apache.iotdb.db.mpp.operator.process.ProcessOperator;
+import org.apache.iotdb.db.mpp.sql.planner.plan.node.PlanNodeId;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+import org.apache.iotdb.tsfile.read.common.block.TsBlock;
+import org.apache.iotdb.tsfile.read.common.block.TsBlockBuilder;
+import org.apache.iotdb.tsfile.utils.Binary;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+
+public class CountMergeOperator implements ProcessOperator {
+  private final PlanNodeId planNodeId;
+  private final OperatorContext operatorContext;
+  private boolean isFinished;
+
+  private final List<Operator> children;
+
+  public CountMergeOperator(
+      PlanNodeId planNodeId, OperatorContext operatorContext, List<Operator> children) {
+    this.planNodeId = planNodeId;
+    this.operatorContext = operatorContext;
+    this.children = children;
+  }
+
+  @Override
+  public OperatorContext getOperatorContext() {
+    return operatorContext;
+  }
+
+  @Override
+  public TsBlock next() {
+    isFinished = true;
+    if (children.get(0) instanceof NodeTimeSeriesCountOperator) {
+      return nextWithGroupByLevel();
+    }
+    return nextWithoutGroupByLevel();
+  }
+
+  private TsBlock nextWithoutGroupByLevel() {
+    CountDownLatch countDownLatch = new CountDownLatch(children.size());
+    List<TsBlock> tsBlocks = new ArrayList<>(children.size());
+    for (Operator child : children) {
+      new Thread(
```

Comment:
According to our design, each operator should only be run inside one Thread and cannot start new uncontrolled threads.

Another issue is, if the main thread is blocked by its children, it should return Blocked by `isBlock()` rather than waiting its children

Reply:
Many thanks. I found I have misunderstood the execution procedure of mpp until I read the code of `SinkHandle` and `SourceHandle`, I will fix it soon.